### ISSUE TITLE
Download all overwrites existing files with same name

### ISSUE THREAD

**Issue by aryehdesigns** [nylas/N1#2980]
##### What operating system are you using?
Mac OSX 10.11.6

##### What version of N1 are you using?
0.4.59-cae9ef1

--

**Bug**

When using the `Download all` button to download an email's attachments, it will overwrite any existing files with the same name (within the destination folder). These 'old' files also don't appear in Trash, so I have no way of recovering them after they have been overwritten. I've now fallen into this trap a few times and realised too late resulting in lost documents.

Either the user should be made aware of the overwrite and be able to cancel before it takes place, or the new file should automatically have a `-1` (or similar) appended to the filename.

**Comment by molstrangler** [comment]
This also happens on the Windows client..

The issue is the image is saved with exactly the same file name, so each file is overwritten by the next.  Hence you not seeing any images in the Trash.

**Comment by aryehdesigns** [comment]
Yeah, I figured something like that was happening. Quite a dangerous bug imo... Hope it gets picked up by the dev team quickly.

**Comment by molstrangler** [comment]
I'm not sure its a destructive bug, your images still exist in email, so you can save one at a time and manually choose a different file name. But it's still a bug!

There are kinda silly mistakes that get through.  Like being able to assign the same key to different shortcuts in preferences, it's a good insight into how their logic flows.

**Comment by aryehdesigns** [comment]
Hmm not sure I agree, I think it’s actually quite a destructive bug!

Last week I received a PDF from a client which I saved and edited locally (added annotations, etc). Then this afternoon I received a new version of the PDF which had the same name as last week (of course I didn't realise this) and after saving it locally in the client's folder, the file got overwritten, meaning I've lost all my edits from last week. Not great...

**Comment by molstrangler** [comment]
I didn't realize this bug affected all attachments, I thought it was how they handled images.

Then it is more serious than I understood.

An alternative could be for the email client to reveal the original file in a finder or explorer window instead of copying the file to another location on your hard disk.  Then you can copy & paste the file to where every you wanted.

It would simplify their code for sure.

**Comment by aryehdesigns** [comment]
Oh right, yeah it's happened with numerous different file types now. Definitely hope it'll get fixed one way or another soon.

For now I'm just gonna save everything to my Downloads folder and move things manually; a few extra steps but a lot safer.
